spacewalk/admin/salt-secrets-config.py

```diff
@@ -60,5 +65,7 @@
     os.chmod("/etc/salt/pki/api/salt-api.crt", 0o600)
     os.chown("/etc/salt/pki/api/salt-api.key", pwd.getpwnam("salt").pw_uid, grp.getgrnam("salt").gr_gid)
     os.chmod("/etc/salt/pki/api/salt-api.key", 0o600)
-    shutil.copyfile("/etc/salt/pki/api/salt-api.crt", "/etc/pki/trust/anchors/salt-api.crt")
-    os.system("update-ca-certificates")
+    shutil.copyfile("/etc/salt/pki/api/salt-api.crt", cert_location + "/salt-api.crt")
+    if os.system("update-ca-certificates"):
+        print("update-ca-certificates failed. Perform the required activities manually.")
```

Comment:
I think on CentOS you need to use this tool `/usr/bin/update-ca-trust`, right?
Why not putting it into another variable and enhance the check above to set also the ca tool to call?

Reply:
Thanks, I didn't know that. Will check it out and update.

Reply:
Both tools seem to do a similar thing, so I added update-ca-trust. Before a final release, we really have to go through those hard coded /etc/pki folders and match them up...
I did follow a different strategy to your suggestion. Instead of trying to complicatedly detect the right tool at the beginning, I run one tool. If it fails, the other one is run. This keeps the code clean from one time variables and complex logic.